runtime/src/main/java/org/corfudb/runtime/view/LayoutView.java

```diff
@@ -295,11 +295,12 @@ public void committed(long epoch, Layout layout, boolean force)
 
         int timeouts = 0;
         int responses = 0;
-        while (responses < commitList.length) {
```

Comment:
I am failing to see the bug. You will be iterating while (responses < commitList.length). `CompletableFuture.anyOf()` will be invoked for all CF which do not execute exceptionally.

Reply:
On every iteration only one future can bump up the (via CompletableFuture.anyOf) responses counter.
The response counter should be incremented once per future, but one cf can increment it multiple times, which is the bug.

Reply:
I see.